### PR TITLE
Update List-Detail sample to Material 3 Adaptive

### PR DESCRIPTION
**Pull request by tiwiz**
Update List-Detail sample to Material 3 Adaptive

### CODE REVIEW

**Review by IanGClifton — COMMENTED** [review]
LGTM but probably Alex should also look. One alternative is that the selected word index could be used as the ID directly with the navigator instead of using <Nothing>, but I don't have a strong opinion on whether that's better or just different.

File: CanonicalLayouts/list-detail-compose/app/src/main/java/com/example/listdetailcompose/ui/ListDetailSample.kt [hunk 1]
```diff
@@ -73,70 +70,40 @@ private data class DefinedWord(
...
-        detailKey = selectedWordIndex,
-        list = { isDetailVisible ->
+    ListDetailPaneScaffold(
+        directive = navigator.scaffoldDirective,
+        value = navigator.scaffoldValue,
+        listPane = {
             val currentSelectedWordIndex = selectedWordIndex
             ListContent(
                 words = sampleWords.map(DefinedWord::word),
-                selectionState = if (isDetailVisible && currentSelectedWordIndex != null) {
```
Comment (alexvanyo) [inline CanonicalLayouts/list-detail-compose/app/src/main/java/com/example/listdetailcompose/ui/ListDetailSample.kt:108]: I think removing `isDetailVisible` here will result in showing a "selection" even when just the list is visible, which isn't ideal. I think we can read from the `navigator.scaffoldValue` to derive the equivalent logic of `isDetailVisible`:

```
navigator.scaffoldValue[ListDetailPaneScaffoldRole.Detail] == PaneAdaptedValue.Expanded
```

**Review by alexvanyo — COMMENTED** [review]
Looks good other than the one comment to preserve the selection visibility behavior using `scaffoldValue`.

+1 that we could use the navigator's type to store the id, that'd be a neat follow-up PR to see the diff for just making that refactor.

File: CanonicalLayouts/list-detail-compose/app/src/main/java/com/example/listdetailcompose/ui/ListDetailSample.kt (hunk 1 above)
Comment (tiwiz) [reply CanonicalLayouts/list-detail-compose/app/src/main/java/com/example/listdetailcompose/ui/ListDetailSample.kt:108]: Fixed in the following commit

**Review by alexvanyo — APPROVED** [review]
LGTM!

**Pull request merged**